src/basic/escape.c

```diff
@@ -474,14 +474,21 @@ char* octescape(const char *s, size_t len) {
 static char* strcpy_backslash_escaped(char *t, const char *s, const char *bad) {
         assert(bad);
 
-        for (; *s; s++)
-                if (char_is_cc(*s))
-                        t += cescape_char(*s, t);
-                else {
+        while (*s) {
```

Comment:
`while (s && *s)`

Reply:
The current code already assumes that `t` and `s` are non-`NULL`. Any changes to this, if necessary, should be done in another PR.

Reply:
please add assert(t) and assert(s) to the top. we generally have this rule that when you touch code and it isn't really up to our current coding style, then you fix it while touching it. Here it means encoding expectations explicitly via assert() or similar. Hence please add the two lines, either into this commit or as separate commits in your PR.

assert()s have three uses: they are documentation about expectations, they are safety nets to catch unexpected programming errors, and they are nowadays used by compilers and static analyzers to validate and optimize code, as it encodes in computer readable ways what state is possible and what is deemed impossible.

Reply:
Done.